ibm/resource_ibm_cdn.go

```diff
@@ -0,0 +1,579 @@
+			RespectHeaders:           sl.Bool(respectheaders),
+			CertificateType:          sl.String(certificateType),
+			CacheKeyQueryRule:        sl.String(cachekeyqueryrule),
+			PerformanceConfiguration: sl.String(performanceconfiguration),
+		})
+		if err != nil {
+			return fmt.Errorf("Error creating CDN: %s", err)
+		}
+		///Print the response of the requested the service.
+		log.Print("Response for cdn order")
+		log.Println(receipt5)
+		d.SetId(*receipt5[0].UniqueId)
+		id, err := strconv.Atoi((d.Id()))
+		result5, err := service.VerifyDomainMapping(&id)
+		log.Println(result5)
+		return resourceIBMCDNRead(d, meta)
+	}
+	if origintype == "HOST_SERVER" && protocol == "HTTP_AND_HTTPS" {
+		receipt6, err := service.CreateDomainMapping(&datatypes.Container_Network_CdnMarketplace_Configuration_Input{
+			Origin:                   sl.String(originaddress),
+			VendorName:               sl.String(vendorname),
+			Domain:                   sl.String(domain),
+			Path:                     sl.String(path),
+			Cname:                    sl.String(cname),
+			Protocol:                 sl.String(protocol),
+			HttpPort:                 sl.Int(httpport),
+			HttpsPort:                sl.Int(httpsport),
+			OriginType:               sl.String(origintype),
+			Header:                   sl.String(header),
+			RespectHeaders:           sl.Bool(respectheaders),
+			CertificateType:          sl.String(certificateType),
+			CacheKeyQueryRule:        sl.String(cachekeyqueryrule),
+			PerformanceConfiguration: sl.String(performanceconfiguration),
+		})
+		if err != nil {
+			return fmt.Errorf("Error creating CDN: %s", err)
+		}
+		///Print the response of the requested the service.
+		log.Print("Response for cdn order")
+		log.Println(receipt6)
+		d.SetId(*receipt6[0].UniqueId)
+		id, err := strconv.Atoi((d.Id()))
+		result6, err := service.VerifyDomainMapping(&id)
+		log.Println(result6)
+		return resourceIBMCDNRead(d, meta)
+	}
+
+	return nil
+}
+
+func resourceIBMCDNRead(d *schema.ResourceData, meta interface{}) error {
+	sess := meta.(ClientSession).SoftLayerSession()
+	log.Println("reading cdn service...")
+	service := services.GetNetworkCdnMarketplaceConfigurationMappingService(sess)
+	cdnId := sl.String(d.Id())
+	///read the changes in the remote resource and update in the local resource.
+	read, err := service.ListDomainMappingByUniqueId(cdnId)
+	///Print the response of the requested the service.
+	d.Set("id", *read[0].UniqueId)
+	d.Set("originaddress", *read[0].OriginHost)
+	d.Set("vendorname", *read[0].VendorName)
+	d.Set("domain", *read[0].Domain)
+	d.Set("header", *read[0].Header)
+	d.Set("cname", *read[0].Cname)
+	d.Set("origin_type", *read[0].OriginType)
+	d.Set("status", *read[0].Status)
+
+	log.Print("Response for cdn verification: ")
+
+	if err != nil {
+		log.Println("error Reading")
+		log.Println(err)
+	}
+	return nil
+}
+
+func resourceIBMCDNUpdate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
https://cloud.ibm.com/docs/infrastructure/CDN?topic=CDN-cdn-api-reference#updatedomainmapping as per the above link only few fields can be updated...So based on this the fields which cann't be updated add ForceNew : true in schema defination